ext/gd/gd.c

```diff
@@ -3840,28 +3882,36 @@ PHP_FUNCTION(imagecrop)
 	im = php_gd_libgdimageptr_from_zval_p(IM);
 
 	if ((tmp = zend_hash_str_find(Z_ARRVAL_P(z_rect), "x", sizeof("x") -1)) != NULL) {
-		rect.x = zval_get_long(tmp);
+		if (!php_gd_zval_try_get_c_int(tmp, "x", &rect.x)) {
```

Comment:
Aren't you breaking reference handling now with this change?

Reply:
oh ok that s a subtlety I did not notice between zval_get_long/zval_try_get_long if this is what you mean.

Reply:
Indeed I mean that. It probably even crashes if it were a ref.

Reply:
Your current code should fix the crashes, but since you now deref at all callsites (either by using the _deref variant of the zend_hash API; or by using ZVAL_DEREF; it's probably better to use a single ZVAL_DEREF in php_gd_zval_try_get_c_int)